esmvalcore/_config/_diagnostics.py

```diff
@@ -0,0 +1,138 @@
+        return self.path / 'recipes'
+
+    @property
+    def references(self):
+        """Return location of the references (bibtex files)."""
+        return self.path / 'references'
+
+    @property
+    def tags_config(self):
+        """Return location of the tags config."""
+        return self.path / 'config-references.yml'
+
+    @property
+    def scripts(self):
+        """Return location of diagnostic scripts."""
+        return self.path / 'diag_scripts'
+
+    def load_tags(self):
+        """Load the tags config into an instance of ``TagsManager``."""
+        return TagsManager.from_file(self.tags_config)
+
+    @classmethod
+    def find(cls):
+        """Try to find installed diagnostic scripts."""
+        try:
+            import esmvaltool
+        except ImportError:
+            path = Path.cwd()
+        else:
+            # avoid a crash when there is a directory called
+            # 'esmvaltool' that is not a Python package
+            if esmvaltool.__file__ is None:
+                path = Path.cwd()
+            else:
+                path = Path(esmvaltool.__file__).absolute().parent
+        logger.debug('Using diagnostics from %s', path)
+        return cls(path)
+
+
+class TagsManager(dict):
+    """Tag manager."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.source_file = None
+
+    @classmethod
+    def from_file(cls, filename: str):
+        """Load the reference tags used for provenance recording."""
+        if os.path.exists(filename):
+            logger.debug("Loading tags from %s", filename)
+            with open(filename) as file:
+                tags = cls(yaml.safe_load(file))
+                tags.source_file = filename
+                return tags
+        else:
+            # This happens if no diagnostics are installed
+            logger.debug("No tags loaded, file %s not present", filename)
+            return cls()
+
+    def get_tag_value(self, section: str, tag: str):
+        """Retrieve the value of a tag from a section.
+
+        Parameters
+        ----------
+        section : str
+            Name of the subsection
+        tag : str
+            Name of the tag
+        """
+        if section not in self:
+            postfix = f' in {self.source_file}' if self.source_file else ''
+            raise ValueError(f"Section '{section}' does not exist{postfix}")
+
+        if tag not in self[section]:
+            postfix = f' of {self.source_file}' if self.source_file else ''
+            raise ValueError(
+                f"Tag '{tag}' does not exist in section '{section}'{postfix}")
+
+        return self[section][tag]
+
+    def get_tag_values(self, section: str, tags: tuple):
+        """Retrieve the values for a list of tags from a section.
+
+        Parameters
+        ----------
+        section : str
+            Name of the subsection
+        tag : tuple[str] or list[str]
```

Comment:
```suggestion
        tags : tuple[str] or list[str]
```
Codacy is pointing out that you did not add a description of the `tags` argument.

Reply:
🤦‍♂️

Reply:
I spent an embarrassingly long time trying to resolve this... 🙄